src/main/java/org/jabref/gui/citationkeypattern/GenerateCitationKeyAction.java

```diff
@@ -45,8 +47,14 @@ public void execute() {
 
         checkOverwriteKeysChosen();
 
-        BackgroundTask.wrap(this::generateKeys)
-                      .executeWith(Globals.TASK_EXECUTOR);
+        if (!this.isCanceled) {
+            BackgroundTask backgroundTask = this.generateKeysInBackground();
+            backgroundTask.showToUser(true);
+            backgroundTask.titleProperty().set(Localization.lang("Autogenerate citation keys"));
+            backgroundTask.messageProperty().set(Localization.lang("%0 entries", entries.size()));
+
+            backgroundTask.executeWith(Globals.TASK_EXECUTOR);
```

Comment:
Just one minor thing, would be nice if you could pass the dependency to the taskExecutor through the constructor. We are trying to pass dependencies through the constructor. You can simply pass it from JabRefFrame.